engine/src/main/java/org/hibernate/search/engine/search/query/dsl/SearchQueryOptionsStep.java

```diff
@@ -155,4 +162,14 @@
 	 */
 	S totalHitCountThreshold(long totalHitCountThreshold);
 
+	/**
+	 * Configure the query level highlighting optional settings.
+	 * <p>
+	 * For specifying the fields to highlight and providing field-specific settings see {@link SearchProjectionFactory#highlight(String)}
+	 *
+	 * @return {@code this}, for method chaining.
+	 */
+	@Incubating
+	S highlight(Function<QueryHighlightConfigurationTypeStep, QueryHighlightConfigurationFinalStep> highlightConfigurationContributor);
```

Comment:
Reminder: if we follow my recommendations above, this would become something like:

```suggestion
	S highlighter(Function<SearchHighlighterFactory, HighlighterFinalStep> highlighterContributor);
```